modules/text_server_adv/text_server_adv.cpp

```diff
@@ -6097,21 +6098,33 @@ void TextServerAdvanced::_shape_run(ShapedTextDataAdvanced *p_sd, int64_t p_star
 			gl.index = glyph_info[i].codepoint;
 			if (gl.index != 0) {
 				_ensure_glyph(fd, fss, gl.index | mod);
+				if (subpos) {
+					gl.x_off = (double)glyph_pos[i].x_offset / (64.0 / scale);
+				} else {
+					if (p_sd->orientation == ORIENTATION_HORIZONTAL) {
```

Comment:
I'd merge this else block into an `if else` as it's just nested, it would be more readable